relay-server/src/metrics_extraction/transactions/mod.rs

```diff
@@ -736,6 +740,138 @@ fn extract_captured_substring<'a>(string: &'a str, pattern: &'a Lazy<Regex>) ->
+            if op.starts_with("ui.react") {
+                "ui.react"
+            } else if op.starts_with("ui.vue") {
+                "ui.vue"
+            } else if op.starts_with("ui.svelte") {
+                "ui.svelte"
+            } else if op.starts_with("ui.angular") {
+                "ui.angular"
+            } else if op.starts_with("ui.ember") {
+                "ui.ember"
+            } else {
+                "ui"
+            }
+        }
+        //
+        // Web server
+        // `http*` mapped in Browser
+        // `serialize*` mapped in Browser
+        else if op.starts_with("websocket") {
+            "websocket"
+        } else if op.starts_with("rpc") {
+            "rpc"
+        } else if op.starts_with("grpc") {
+            "grpc"
+        } else if op.starts_with("graphql") {
+            "graphql"
+        } else if op.starts_with("subprocess") {
+            "subprocess"
+        } else if op.starts_with("middleware") {
+            "middleware"
+        } else if op.starts_with("view") {
+            "view"
+        } else if op.starts_with("template") {
+            "template"
+        } else if op.starts_with("event") {
+            "event"
+        } else if op.starts_with("function") {
+            if op.starts_with("function.nextjs") {
+                "function.nextjs"
+            } else if op.starts_with("function.remix") {
+                "function.remix"
+            } else if op.starts_with("function.gpc") {
+                "function.grpc"
+            } else if op.starts_with("function.aws") {
+                "function.aws"
+            } else if op.starts_with("function.azure") {
+                "function.azure"
+            } else {
+                "function"
+            }
+        } else if op.starts_with("console") {
+            "console"
+        } else if op.starts_with("file") {
+            "file"
+        } else if op.starts_with("app") {
+            "app"
+        }
+        //
+        // Database
+        else if op.starts_with("db") {
+            "db"
+        } else if op.starts_with("cache") {
+            "cache"
+        }
+        //
+        // Serverless
+        // `http*` marked in Browser
+        // `grpc*` marked in Web server
+        // `function*` marked in Web server
+        //
+        // Mobile
+        // `app*` marked in Web server
+        // `ui*` marked in Browser
+        // `navigation*` marked in Browser
+        // `file*` marked in Web server
+        // `serialize*` marked in Web server
+        // `http*` marked in Browser
+
+        // Desktop
+        // `app*` marked in Web server
+        // `ui*` marked in Browser
+        // `serialize*` marked in Web server
+        // `http*` marked in Browser
+
+        // Messages / queues
+        else if op.starts_with("topic") {
+            "topic"
+        } else if op.starts_with("queue") {
+            "queue"
+        }
+        //
+        // Unknown
+        else {
+            ""
```

Comment:
I think you still can keep `return None` here, it would be fine.  so you don't have to do another check at the end. 
And at the end just return `Some(category)`.